GAE/src/org/akvo/flow/domain/persistent/Report.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright (C) 2018 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.domain.persistent;
+
+import java.util.Date;
+
+import javax.jdo.annotations.PersistenceCapable;
+
+import com.gallatinsystems.framework.domain.BaseDomain;
+
+/**
+ * domain to store a generated report
+ *
+ */
+@PersistenceCapable
+public class Report extends BaseDomain {
+
+    public static final String QUEUED = "QUEUED";
+    public static final String IN_PROGRESS = "IN_PROGRESS";
+    public static final String FINISHED_SUCCESS = "FINISHED_SUCCESS";
+    public static final String FINISHED_ERROR = "FINISHED_ERROR";
+
+    private static final long serialVersionUID = -1;
+    private String reportType;
+    private Long formId;
+    private String state;
+    private String filename;
+    private String message;
+    private Long user;
```

Comment:
Because the report engine does not get authorized as a user. The user id will be 0; but since the report engine no longer creates the Report entity, it is less of a problem.